### DISCUSSION TITLE
Looking for some help to achieve a specific outcome.

### DISCUSSION THREAD

**Discussion by fraxyl** [vladmandic/sdnext#2807] [Q&A] [answered]
I had an idea to create a music video for a song I've written. So far I've recorded the video of me, processed it using TouchDesigner to create a line art style of the video and then taken that video and saved the frames to images to use in ControlNet.

I have my controlnet set to Canny and batch is pointed at the folder where the frames are, named to the effect of frame##### where # is the frame number. When I set the text2image batch, does it step through the controlnet images as it generates from my text2image prompt?

Also the batch only goes to 100 and I have a little under 5000 frames. Should I just run a batch of 100 each time and remove the frames from the controlnet batch folder as I go or what?

I tried following a youtube tutorial to start with, nearly everything I search for is A1111 or ComfyUI, and attempting to install either of those caused more headaches.

**Comment by brknsoul** [comment, marked as answer]
If you're talking about the Batch Count slider, you can edit that in ui-config.json, by adding a comma to the end of the last line before the `}`, then adding;
`"txt2img/Batch count/maximum": 100`
to the next line, changing the 100 to a larger value.

eg:
![image](https://github.com/vladmandic/automatic/assets/1969381/62f613f1-f06e-4d20-b29a-58c9d56892ff)

**Reply by fraxyl** [reply]
My ui-config.json is just { } is this normal?

**Comment by vladmandic** [comment]
`ui-config.json` only saves values which are different than default. so until modifications are made, its empty. but that's besides the point - not needed.

what you want is to process 3k frames using process batch. why would you need to set batch-size or batch-count for that at all???
setting batch-size or count to 100 would result in producing 100 images for EACH of your input frames.

**Reply by fraxyl** [reply]
Mostly because I'm not sure how to get the result I'm after or what I'm doing so I must experiment, ask questions, learn, adjust, reiterate. :)

Just going on the normal terms for batch within an environment I'm unfamiliar with.

I can put a prompt in and generate a result just fine, I'm just not sure if I'm going about what I want to do in the right way because it's like stumbling around in a dark room. I'm also not sure I'm able to explain my thought process properly. I'm mostly an artist rather than a software developer.

![image](https://github.com/vladmandic/automatic/assets/159160137/573d448c-6870-4feb-99a6-a18a8ed5ec17)

This is sort of what my brain is thinking, I just want to automate this X times. Where X is currently 4590

My batch size is only set to 1, batch count is what I thought I needed to increase, but now I'm not sure, when it was set to 1, I hit generate and it generated 1 image and stopped. Process gives no options for generating an image based on prompt.

**Reply by vladmandic** [reply]
and where would 4590 prompts come from?

**Reply by fraxyl** [reply]
It's only one prompt, the controlnet is what changes.

**Reply by vladmandic** [reply]
then process -> process folder is what you want.
process batch would mean that 4590 images need to be uploaded and that would not work.

**Reply by fraxyl** [reply]
process does not appear to achieve what I had in mind, I set the batch count to 4590 and it appears to be doing what I asked. I haven’t uploaded 4590 images, that number of images is just sitting in a folder that I pointed to in the batch section of controlnet. from my understanding, the process tab is for processing images after they have been generated, is this not the case?

**Reply by vladmandic** [reply]
Right.
You can use new control tab for that. It has both functionalities that you need.
It's in dev now, release likely tomorrow.

**Reply by fraxyl** [reply]
After a few tries, I think I have my workflow worked out, but now I'm not sure that the batch section of control net is working how I understand it to work. When I point it to a folder on my computer, what is controlnet expecting to find? Does it use the subdirectories inside? I've managed to reduce the amount of controlnet images I want to use from 4k to 1100, but I noticed when I changed the folder in the batch section, it started producing the same images that it generated the first time. Another time I had the batch pointed at a folder that no longer existed and it continued to generate images no problem.
